File: 01/sol.py
from collections import Counter

t = [map(int, l.split()) for l in open("inp.txt").readlines()]

a, b = zip(*t)
a = sorted(list(a))
b = sorted(list(b))

print(sum(abs(p1 - p2) for p1, p2 in zip(a, b)))

temp = Counter(b)

print(sum(n * temp[n] for n in a))

File: 01/sol.hs
import Data.List
import Data.Map hiding (map)

tup2 [a, b] = (a, b)

main :: IO ()
main = do
  contents <- readFile "inp.txt"
  let (a,b) = unzip $
       map (tup2 . (map read) . words) $ lines contents
      c = fromListWith (+) $ map (,1) b
  print $
    sum $ map (abs . (uncurry (-))) $ zip (sort a) (sort b)
  print $
    sum $ map (\n -> n * findWithDefault 0 n c) a

